dev-docs/component-dev-notes.md

```diff
@@ -0,0 +1,30 @@
+### Errors
+
+Components are intended to interface with the registry system - they are registratable objects.
+The intent of this is that the system knows what objects have been registered so it can present to the user on an API level all the components which can be included in a bot configuration.
+However, setting this up does mean adding some definitions in a couple of different places.
+
+### If you are changing a declared API
```

Comment:
Probably want to consider a "if you're changing an API, increment the verno as per this scheme"; or at least a reference here to the other document which says how that's done. An api-dev-notes.md?

Reply:
I have added a reference to api-dev-notes.md - will construct.